examples/mps3-an536/src/bin/generic_timer.rs

```diff
@@ -17,14 +17,8 @@ pub extern "C" fn kmain() {
     semihosting::process::exit(0);
 }
 
-/// A placeholder function so that the workspace still builds
-#[cfg(not(arm_architecture = "v8-r"))]
-fn main() {
-    println!("No generic timers on this platform");
-}
-
 /// Let's test some timers!
-#[cfg(arm_architecture = "v8-r")]
+#[export_name = "main"]
```

Comment:
Why is this unmangled?

Reply:
On the Zynq target, I had issues setting through code when the main method was not explicitely exported as "main". I have not tried stepping through the code with QEMU, but I assumed this is just generally a good idea. This is also something that is done by the cortex-m-rt entry point macro.